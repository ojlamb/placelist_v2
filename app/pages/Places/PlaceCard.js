import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';


const styles = {
  card: {
    maxWidth: 345
  },
  media: {
    height: 140
  },
  link: {
    textDecoration: 'none',
  }
};


const getCardImage = data => require(`common/images/${data}.jpg`);


export class PlaceCard extends React.PureComponent {
  render() {
    const { classes } = this.props;
    return (
      <Card className={classes.card}>
        <CardMedia
          className={classes.media}
          image={getCardImage(this.props.category)}
          title={this.props.name}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {this.props.name}
          </Typography>
          <Typography component="p">{this.props.address}</Typography>
        </CardContent>
        <CardActions>
          <Link className={classes.link} to={`/place/${this.props.id}`}>
            <Button size="small" color="primary">
              More
            </Button>
          </Link>
        </CardActions>
      </Card>
    );
  }
}
PlaceCard.propTypes = {
  address: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired
};

export default withStyles(styles)(PlaceCard);
