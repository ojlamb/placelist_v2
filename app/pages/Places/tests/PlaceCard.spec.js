import React from 'react';
import { PlaceCard } from '../PlaceCard';

const getProps = ({
  address,
  category,
  id,
  name,
} = {}) => ({
  classes: {},
  address,
  category,
  id,
  name,
});

const getJSX = props => (<PlaceCard {...getProps(props)} />);

describe('HeaderView', () => {
  describe('snapshots', () => {
    const scenarios = [
      { address: '123 main st.',
        category: 'dinner',
        id: 1,
        name: 'Owens Grill'
      }
    ];

    scenarios.forEach((props, i) => {
      describe(`Scenario ${i + 1}`, () => {
        it('matches snapshot', () => {
          const wrapper = shallow(getJSX(props));
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});
