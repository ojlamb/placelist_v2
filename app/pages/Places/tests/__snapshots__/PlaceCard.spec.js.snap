// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderView snapshots Scenario 1 matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlaceCard
    address="123 main st."
    category="dinner"
    classes={Object {}}
    id={1}
    name="Owens Grill"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardMedia)
          image="IMAGE_MOCK"
          title="Owens Grill"
        />,
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            Owens Grill
          </WithStyles(Typography)>
          <WithStyles(Typography)
            component="p"
          >
            123 main st.
          </WithStyles(Typography)>
        </WithStyles(CardContent)>,
        <WithStyles(CardActions)>
          <Link
            replace={false}
            to="/place/1"
          >
            <WithStyles(Button)
              color="primary"
              size="small"
            >
              More
            </WithStyles(Button)>
          </Link>
        </WithStyles(CardActions)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": undefined,
          "image": "IMAGE_MOCK",
          "title": "Owens Grill",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              Owens Grill
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              component="p"
            >
              123 main st.
            </WithStyles(Typography)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Owens Grill",
              "component": "h2",
              "gutterBottom": true,
              "variant": "h5",
            },
            "ref": null,
            "rendered": "Owens Grill",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "123 main st.",
              "component": "p",
            },
            "ref": null,
            "rendered": "123 main st.",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            replace={false}
            to="/place/1"
          >
            <WithStyles(Button)
              color="primary"
              size="small"
            >
              More
            </WithStyles(Button)>
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              size="small"
            >
              More
            </WithStyles(Button)>,
            "className": undefined,
            "replace": false,
            "to": "/place/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "More",
              "color": "primary",
              "size": "small",
            },
            "ref": null,
            "rendered": "More",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardMedia)
            image="IMAGE_MOCK"
            title="Owens Grill"
          />,
          <WithStyles(CardContent)>
            <WithStyles(Typography)
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              Owens Grill
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
            >
              123 main st.
            </WithStyles(Typography)>
          </WithStyles(CardContent)>,
          <WithStyles(CardActions)>
            <Link
              replace={false}
              to="/place/1"
            >
              <WithStyles(Button)
                color="primary"
                size="small"
              >
                More
              </WithStyles(Button)>
            </Link>
          </WithStyles(CardActions)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": undefined,
            "image": "IMAGE_MOCK",
            "title": "Owens Grill",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                component="h2"
                gutterBottom={true}
                variant="h5"
              >
                Owens Grill
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                component="p"
              >
                123 main st.
              </WithStyles(Typography)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Owens Grill",
                "component": "h2",
                "gutterBottom": true,
                "variant": "h5",
              },
              "ref": null,
              "rendered": "Owens Grill",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "123 main st.",
                "component": "p",
              },
              "ref": null,
              "rendered": "123 main st.",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/place/1"
            >
              <WithStyles(Button)
                color="primary"
                size="small"
              >
                More
              </WithStyles(Button)>
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                size="small"
              >
                More
              </WithStyles(Button)>,
              "className": undefined,
              "replace": false,
              "to": "/place/1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "More",
                "color": "primary",
                "size": "small",
              },
              "ref": null,
              "rendered": "More",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
